src/ctapipe/io/interpolation.py

```diff
@@ -0,0 +1,283 @@
+
+        Parameters
+        ----------
+        tel_id : int
+            Telescope id
+        input_table : astropy.table.Table
+            Table of pointing values, expected columns
+            are ``time`` as ``Time`` column, ``azimuth`` and ``altitude``
+            as quantity columns for pointing and pointing correction data.
+        """
+
+        self._check_tables(input_table)
+
+        if not isinstance(input_table["time"], Time):
+            raise TypeError("'time' column of pointing table must be astropy.time.Time")
+
+        input_table.sort("time")
+
+        az = input_table["azimuth"].quantity.to_value(u.rad)
+        # prepare azimuth for interpolation by "unwrapping": i.e. turning
+        # [359, 1] into [359, 361]. This assumes that if we get values like
+        # [359, 1] the telescope moved 2 degrees through 0, not 358 degrees
+        # the other way around. This should be true for all telescopes given
+        # the sampling speed of pointing values and their maximum movement speed.
+        # No telescope can turn more than 180° in 2 seconds.
+        az = np.unwrap(az)
+        alt = input_table["altitude"].quantity.to_value(u.rad)
+        mjd = input_table["time"].tai.mjd
+        self._interpolators[tel_id] = {}
+        self._interpolators[tel_id]["az"] = interp1d(mjd, az, **self.interp_options)
+        self._interpolators[tel_id]["alt"] = interp1d(mjd, alt, **self.interp_options)
+
+
+class CalibrationInterpolator(Interpolator):
+    """
+    Interpolator for calibration data
+    """
+
+    table_location = "dl1/calibration"  # TBD
+
+    def __call__(self, tel_id, time):
+        """
+        Interpolate pedestal or gain for a given time and tel_id.
+
+        Parameters
+        ----------
+        tel_id : int
+            telescope id
+        time : astropy.time.Time
+            time for which to interpolate the calibration data
+
+        Returns
+        -------
+        cal : array [float]
+            interpolated calibration quantity
+        """
+
+        self._check_interpolators(tel_id)
+
+        cal = self._interpolators[tel_id][self.par_name](time)
+        return cal
+
+    def add_table(self, tel_id, input_table, par_name="pedestal"):
+        """
+        Add a table to this interpolator
+
+        Parameters
+        ----------
+        tel_id : int
+            Telescope id
+        input_table : astropy.table.Table
+            Table of pointing values, expected columns
+            are ``time`` as ``Time`` column. The calibration
+            parameter column is given through the variable ``par_name``
+        par_name : str
+            Name of the parameter that is to be interpolated
+            ``pedestal`` is used for pedestals, ``gain`` for gain
+            can also be the name of statistical parameters to
+            interpolate the content of StatisticsContainers
+        """
+
+        self.par_name = par_name
+
+        self.required_columns = {"time", par_name}
```

Comment:
this shadows the class variable `Interpolator.required_columns` with an instance variable of the same name.  Is that intended? I would expect it to cause unexpected behavior/bugs in the future...  (maybe more argument for not using class variables for this)

Reply:
No, certainly not. 

At minimum, it should extend the class variable, not shadow it.

I also don't understand how this "parname" is supposed to work at all.

In the pointing interpolator, `add_table` sets up all necessary interpolators. Here, it seems, it only sets up one column per call of `add_table`, which I think is not how it should be.


`add_table` should setup all interpolators relevant for this class.